src/parse.py

```diff
@@ -40,6 +40,25 @@
     # Regardless of the above case, here's how the file name should terminate
     '(\s|$|:)+'
 )))
+MASTER_REGEX_WITH_SPACES = re.compile(''.join((
+    #begin the capture
+    '(',
+    #a leading / for absolute dirs if its there
+    '\/?',
+    # now we look at directories, where we allow either normal chars
+    # or a single whitespace char followed by legit chars
+    '(([a-z.A-Z0-9\-_]|\s[a-zA-Z0-9\-_])+\/)+',
```

Comment:
Oh sorry I should clarify -- the normal chars or single white space followed by the normal chars is the **inner match** of the directories, and we allow any number of inner matches for a given directory.

So you can have multiple spaces per folder name, you just can't end in a space or have two spaces in a row. It was a tricky hack since otherwise we would gobble up white space before any / and consider that a directory >___<

If you want you could send over a PR that adds testing or I can try to get to it this week!